UML editor/src/main/java/codecain/UMLMethods.java

```diff
@@ -1,225 +1,170 @@
         }
     }
 
     /**
-     * Adds a parameter to an existing method.
+     * Adds a parameter to a method.
      *
-     * @param className the name of the class
-     * @param methodName the name of the method to add a parameter to
-     * @param parameter the parameter to add
+     * @param className   The name of the class
+     * @param methodName  The name of the method to add a parameter to
+     * @param parameter   The new parameter to add
      */
-    public void addParameter(Object className, Object methodName, Object parameter) {
-        if (isClassNameOrMethodNameOrParameterInvalid(className, methodName, parameter)) return;
-        if (!doesMethodExist(className, methodName)) return;
-        List<Object> parameters = classMethods.get(className).get(methodName);
-        if (!parameters.contains(parameter)) {
-            parameters.add(parameter);
+    public void addParameter(String className, String methodName, String parameter) {
+        UMLClassInfo classInfo = getClassInfo(className);
+        if (classInfo == null) return;
+        UMLMethodInfo method = classInfo.getMethodByName(methodName);
+        if (method == null) {
+            System.out.println("Action Canceled: Method " + methodName + " does not exist in class " + className);
+            return;
+        }
+        if (!method.getParameters().contains(parameter)) {
+            method.addParameter(parameter);
             System.out.println("Parameter " + parameter + " added to method " + methodName + " in class " + className);
         } else {
             System.out.println("Action Canceled: Parameter " + parameter + " already exists in method " + methodName);
         }
     }
 
     /**
-     * Removes a parameter from an existing method.
+     * Removes a parameter from a method.
      *
-     * @param className the name of the class
-     * @param methodName the name of the method to remove the parameter from
-     * @param parameter the parameter to remove
+     * @param className  The name of the class
+     * @param methodName The name of the method to remove a parameter from
+     * @param parameter  The parameter to remove
      */
-    public void removeParameter(Object className, Object methodName, Object parameter) {
-        if (isClassNameOrMethodNameOrParameterInvalid(className, methodName, parameter)) return;
-        if (!doesMethodExist(className, methodName)) return;
-        List<Object> parameters = classMethods.get(className).get(methodName);
-        if (!parameters.contains(parameter)) {
-            System.out.println("Action Canceled: Parameter " + parameter + " does not exist in method " + methodName);
-        } else {
-            parameters.remove(parameter);
+    public void removeParameter(String className, String methodName, String parameter) {
+        UMLClassInfo classInfo = getClassInfo(className);
+        if (classInfo == null) return;
```

Comment:
Improved, using String for clarity. Simplified Logic,  and better error messages. I also like the feedback it adds to the user